servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncPublisherOperator.java

```diff
@@ -59,21 +59,22 @@ public String toString() {
     };
 
     private final Publisher<T> original;
-    private final Supplier<BooleanSupplier> shouldOffloadSupplier;
+    private final Supplier<? extends BooleanSupplier> shouldOffloadSupplier;
     private final Executor executor;
 
     TaskBasedAsyncPublisherOperator(final Publisher<T> original,
-                                    final Supplier<BooleanSupplier> shouldOffloadSupplier, final Executor executor) {
+                                    final Supplier<? extends BooleanSupplier> shouldOffloadSupplier,
+                                    final Executor executor) {
         this.original = original;
         this.shouldOffloadSupplier = shouldOffloadSupplier;
         this.executor = executor;
     }
 
-    BooleanSupplier shouldOffload() {
-        return shouldOffloadSupplier.get();
+    final BooleanSupplier shouldOffload() {
```

Comment:
nit: consider renaming (e.g. `shouldOffloadSupplier`) to avoid overloaded naming with the other internal `shouldOffload`